lib/VPackDeserializer/enum.h

```diff
@@ -39,25 +40,31 @@ struct enum_deserializer {
                 "members of the enum?");
 };
 
-template <typename Enum, typename... Values, Enum... EnumValues>
-struct enum_deserializer<Enum, enum_member<EnumValues, Values>...> {
-  using plan = enum_deserializer<Enum, enum_member<EnumValues, Values>...>;
+template <typename Enum, typename... Values, Enum... EnumValues, bool... IsHidden>
+struct enum_deserializer<Enum, enum_member<EnumValues, Values, IsHidden>...> {
+  using plan = enum_deserializer<Enum, enum_member<EnumValues, Values, IsHidden>...>;
   using constructed_type = Enum;
   using factory = utilities::identity_factory<Enum>;
 };
 
 }  // namespace deserializer
 
 namespace deserializer::executor {
-template <typename Enum, typename... Values, Enum... EnumValues, typename H>
-struct deserialize_plan_executor<enum_deserializer<Enum, enum_member<EnumValues, Values>...>, H> {
+template <typename Enum, typename... Values, Enum... EnumValues, bool... IsHidden, typename H>
+struct deserialize_plan_executor<enum_deserializer<Enum, enum_member<EnumValues, Values, IsHidden>...>, H> {
   using value_type = Enum;
   using tuple_type = std::tuple<value_type>;
   using result_type = result<tuple_type, deserialize_error>;
 
-  template <typename V, typename... Vs>
-  static std::string joinValues() {
-    return to_string(V{}) + ((", " + to_string(Vs{})) + ...);
+  template<typename...>
+  struct type_list{};
+  template<typename, bool>
+  struct value_hidden_pair {};
+
+  template <typename V, bool hidden, bool... hiddens, typename... Vs>
+  static std::string joinValues(type_list<value_hidden_pair<V, hidden>, value_hidden_pair<Vs, hiddens>...>) {
+    static_assert(!hidden, "please make the first entry not hidden");
+    return to_string(V{}) + ((hiddens ? (", " + to_string(Vs{})) : std::string{}) + ...);
```

Comment:
Now this seems to say that only hidden entries are printed (except for the very first argument). Please verify, but I think this should be:
```suggestion
    return to_string(V{}) + ((hiddens ? std::string{} : (", " + to_string(Vs{}))) + ...);
```